modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -785,15 +800,25 @@ private void defineServiceInitFunction(BLangService service, SymbolEnv conEnv) {
 //        service.symbol.initFunctionSymbol = service.initFunction.symbol;
     }
 
-    private BLangAssignment createAssignmentStmt(BLangVariable variable) {
-        BLangAssignment assignmentStmt = (BLangAssignment) TreeBuilder.createAssignmentNode();
-        assignmentStmt.expr = variable.expr;
-        assignmentStmt.pos = variable.pos;
+    private StatementNode createAssignmentStmt(BLangVariable variable) {
         BLangSimpleVarRef varRef = (BLangSimpleVarRef) TreeBuilder
                 .createSimpleVariableReferenceNode();
         varRef.pos = variable.pos;
         varRef.variableName = variable.name;
         varRef.pkgAlias = (BLangIdentifier) TreeBuilder.createIdentifierNode();
+
+        //TODO temporary solution to avoid endpoint assignment until endpoint model changed.
+        if (variable.typeNode.getKind() == NodeKind.ENDPOINT_TYPE) {
```

Comment:
Let's remove this check :)

Reply:
not clear what you meant here :) (whether to treat all the statements as bind statements or allow endpoint assignment) so will allow endpoint assignment for the time being and will remove this check :)